app/components/primer/select_menu_component.rb

```diff
@@ -0,0 +1,550 @@
+        @divider_classes = divider_classes
+        @system_arguments = system_arguments
+        @system_arguments[:tag] ||= :button
+        @system_arguments[:role] ||= if @selected || @icon
+          "menuitemcheckbox"
+        else
+          "menuitem"
+        end
+        @system_arguments[:classes] = class_names(
+          "SelectMenu-item",
+          system_arguments[:classes],
+        )
+        @system_arguments[:"aria-checked"] = "true" if @selected
+      end
+
+      def wrapper_component
+        case @system_arguments[:tag]
+        when :button
+          Primer::ButtonComponent.new(**@system_arguments)
+        when :a
+          Primer::LinkComponent.new(**@system_arguments)
+        else
+          Primer::BaseComponent.new(**@system_arguments)
+        end
+      end
+
+      # Private: Only used if `icon` is non-nil.
+      def icon_component
+        Primer::OcticonComponent.new(
+          icon: icon,
+          classes: class_names(
+            "SelectMenu-icon SelectMenu-icon--check",
+            @icon_classes,
+          )
+        )
+      end
+
+      # Private: Only used if `divider` is non-nil.
+      def divider_component
+        Primer::BaseComponent.new(
+          tag: divider.is_a?(String) ? :div : :hr,
+          classes: class_names("SelectMenu-divider", @divider_classes)
+        )
+      end
+    end
+
+    # An optional header for the select menu.
+    class Header < Primer::Slot
+      DEFAULT_CLOSEABLE = false
+
+      # @param closeable [Boolean] Whether to include a close button in the header for closing the whole menu.
+      # @param title_tag [Symbol] HTML element type for the `.SelectMenu-title` tag; defaults to `:h3`.
+      # @param title_classes [String] CSS classes to apply to the title element within the header.
+      # @param close_button_classes [String] CSS classes to apply to the close button within the header; only used if `closeable` = `true`.
+      # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>, including: `tag` (`Symbol`) - HTML element type for the header tag; defaults to `:header`.
+      def initialize(
+        closeable: DEFAULT_CLOSEABLE,
+        title_tag: :h3,
+        title_classes: nil,
+        close_button_classes: nil,
+        **system_arguments
+      )
+        @closeable = fetch_or_fallback_boolean(closeable, DEFAULT_CLOSEABLE)
+        @title_tag = title_tag
+        @title_classes = title_classes
+        @close_button_classes = close_button_classes
+        @system_arguments = system_arguments
+        @system_arguments[:tag] ||= :header
+        @system_arguments[:classes] = class_names(
+          "SelectMenu-header",
+          system_arguments[:classes]
+        )
+      end
+
+      def closeable?
+        @closeable
+      end
+
+      def wrapper_component
+        Primer::BaseComponent.new(**@system_arguments)
+      end
+
+      def title_component
+        Primer::BaseComponent.new(
+          tag: @title_tag || :h3,
```

Comment:
We also have this fallback to `:h3` in the initializer. Could we consolidate it to a constant or otherwise simplify the fallback behavior to a single location?